### PR TITLE
Update tests to use newer molecule version

### PR DESCRIPTION
**Pull request by joelnb**
Update the tests to the current version of molecule and ensure they are working correctly.

This changes a few things which are worth noting:
- Slightly different list of OSes used for testing. The new list seems reasonable to me but happy to discuss.
- Tests now use python3.
- Tests now only run on a single version of ansible. This also feels reasonable - recent versions of molecule require at least ansible 2.8 so we can't easily test versions before that anyway.

### CODE REVIEW

**Comment by joelnb** [comment]
I can't work out how to make this a draft now I have already created it but I'm still working on a couple of things:

- [ ] Enable yamllint
- [ ] Ensure supported OSes for galaxy matches the versions tested

I haven't yet marked this as closing https://github.com/caddy-ansible/caddy-ansible/issues/2 but if it seems reliable then I may well do that. I may make a separate PR to add the functionality I was talking about in that issue (allowing to specify a github token for API access).

**Comment by joelnb** [comment]
@lbischof if you have a chance to review this at some point it would be much appreciated - it touches quite a few places so a second pair of eyes can't hurt. It seems to be working pretty reliably at the moment.

I've got the changes ready to go to enable supplying a github token & I'll make a separate PR for that once this goes in (because I've rebased that onto this so that they are compatible).

File: molecule/default/converge.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+- name: Converge
+  hosts: all
+  roles:
+    - role: caddy-ansible
+  tasks:
+    - name: Ensure ss is installed (for testinfra)
+      yum:
```
Comment (lbischof) [inline molecule/default/converge.yml:8]: Can we use `package` and remove the condition?

File: tasks/main.yml [hunk 2]
```diff
@@ -26,7 +28,9 @@
 
 - name: Build base URL
   set_fact:
-    caddy_base_url: "https://{{ caddy_license_account_id }}:{{ caddy_license_api_key }}@caddyserver.com/download/linux/{{ go_arch }}"
+    caddy_base_url: "https://{{ caddy_auth_info }}@caddyserver.com/download/linux/{{ go_arch }}"
+  vars:
+    caddy_auth_info: "{{ caddy_license_account_id }}:{{ caddy_license_api_key }}"
```
Comment (lbischof) [inline tasks/main.yml:33]: Maybe `caddy_base_url` and `caddy_auth_info` should be moved into the `defaults.yml` or `vars/main.yml` file?

File: tasks/main.yml [hunk 3]
```diff
@@ -69,8 +73,9 @@
   register: gpg_list_keys
   when: caddy_pgp_verify_signatures
   tags: skip_ansible_lint
-  failed_when: "gpg_list_keys is failed and ('gpg: error reading key: No public key' not in gpg_list_keys.stderr and
-                                             'gpg: error reading key: public key not found' not in gpg_list_keys.stderr)"
+  failed_when: "gpg_list_keys is failed and
```
Comment (lbischof) [inline tasks/main.yml:76]: Would this work?:
```
failed_when: 
  - gpg_list_keys is failed
  - 'gpg: error reading key: No public key' not in gpg_list_keys.stderr
  - 'gpg: error reading key: public key not found' not in gpg_list_keys.stderr
```

File: tasks/packages-apt.yml [hunk 4]
```diff
@@ -2,15 +2,20 @@
...
+    update_cache: true
+    cache_valid_time: 43200   # 12 hours
 
 - name: Install git
   apt:
     name: git
     state: present
   when: caddy_features is search('git')
 
+- name: Install procps
```
Comment (lbischof) [inline tasks/packages-apt.yml:14]: Why is this needed?

File: .travis.yml [hunk 5]
```diff
@@ -1,20 +1,21 @@
...
+  matrix:
+    - MOLECULE_DISTRO: centos7
+    - MOLECULE_DISTRO: centos8
+    - MOLECULE_DISTRO: fedora29
+    - MOLECULE_DISTRO: ubuntu1804
+    - MOLECULE_DISTRO: ubuntu2004
+    - MOLECULE_DISTRO: debian9
+    - MOLECULE_DISTRO: debian10
 install:
-  - pip install ansible==${ANSIBLE} ansible-lint>=4.0.0 molecule==1.25.0 docker git-semver testinfra>=1.7.0
```
[the remark this answers is not in the document]
Comment (joelnb) [reply .travis.yml:11]: Yes but now you mention I will add it back here - seems nice to be explicit.

File: tasks/packages-apt.yml (hunk 4 above)
Comment (joelnb) [reply tasks/packages-apt.yml:14]: The Debian 10 docker image doesn't come with this package (which provides the `/bin/kill` command used in the systemd service file for reloading caddy). I thought about just ensuring it was installed for the test but if it's genuinely needed for the role then it seemed better to just put it here.

File: tasks/main.yml (hunk 3 above)
Comment (joelnb) [reply tasks/main.yml:76]: Had a quick read & it seems like this will do the same thing with much cleaner syntax so I'll update it to use this & retest. Thanks for the suggestion.

File: tasks/packages-apt.yml (hunk 4 above)
Comment (joelnb) [reply tasks/packages-apt.yml:14]: Perhaps I should also ensure the package which provides this on other distros is installed too though?

File: molecule/default/converge.yml (hunk 1 above)
Comment (joelnb) [reply molecule/default/converge.yml:8]: I'm not sure that is possible because the package is named differently (it's iproute2 on Debian-based distros & also on Arch linux). I checked the docs and couldn't find a way to handle this without setting a variable to the name of the package & I'm not sure how it can easily be set depending on the OS in this context without using a conditional `set_fact` (which doesn't feel much of an improvement on the current solution). Do you know a nicer way to handle this?

Guess I should at least ensure it is installed for the other distros for consistency though.

File: tasks/packages-apt.yml (hunk 4 above)
Comment (lbischof) [reply tasks/packages-apt.yml:14]: Ok, then please add a comment why procps is needed. It's weird this is needed though, I thought `kill` is a command that is already installed on pretty much every distro.

File: molecule/default/converge.yml (hunk 1 above)
Comment (lbischof) [reply molecule/default/converge.yml:8]: Ok, thats fine. You can leave it.

File: molecule/default/converge.yml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+---
+- name: Converge
+  hosts: all
+  roles:
+    - role: caddy-ansible
+  tasks:
+    - name: Ensure ss is installed (for testinfra)
```
[the remark this answers is not in the document]
Comment (joelnb) [reply molecule/default/converge.yml:7]: Possibly I am missing something obvious but what exactly is the typo here? I'm just not seeing it!

File: tasks/main.yml (hunk 2 above)
Comment (joelnb) [reply tasks/main.yml:33]: Have done this for `caddy_auth_info` (into `vars/main.yml`).

I didn't do it for `caddy_base_url` because it seemed to end up with a pretty huge jinja2 statement inside the vars (due to it being different depending on whether a license is provided).

File: tasks/packages-apt.yml (hunk 4 above)
Comment (joelnb) [reply tasks/packages-apt.yml:15]: It's a builtin in bash so you can use it without this package being installed (assuming you are in  bash) but the systemd service needs the actual binary.

I was also surprised it's not included tbh.

File: molecule/default/converge.yml (hunk 1 above)
Comment (joelnb) [reply molecule/default/converge.yml:8]: I had already added it for the other distros before I saw this but can remove if you prefer.

**Comment by joelnb** [comment]
And now the tests failed with the rate limit issue so maybe I do need to think some more on that!

https://travis-ci.org/github/caddy-ansible/caddy-ansible/jobs/671557220

File: molecule/default/converge.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (lbischof) [reply molecule/default/converge.yml:7]: I thought `ss` was wrong. Is that a program from `iproute`? Then you can ignore my comment, sorry...

File: molecule/default/converge.yml (hunk 1 above)
Comment (lbischof) [reply molecule/default/converge.yml:8]: If you already added it, then that's also ok :) I just meant, that you can ignore my comment about `package`

File: tasks/packages-apt.yml (hunk 4 above)
Comment (lbischof) [reply tasks/packages-apt.yml:15]: Ah, that makes more sense :) Thanks for clarifying. You can decide regarding the other distros

File: molecule/default/converge.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (joelnb) [reply molecule/default/converge.yml:7]: Yeah it's basically the newer replacement for `netstat` - it serves a similar purpose.

File: tasks/packages-apt.yml (hunk 4 above)
Comment (joelnb) [reply tasks/packages-apt.yml:15]: On RedHat-based distros & Arch this is provided by `util-linux`. I think it's reasonable to assume that is always installed so I'm going to leave this as apt-specific.
